packages/eslint-plugin/src/rules/naming-convention.ts

```diff
@@ -197,8 +205,24 @@ function selectorSchema(
 ): JSONSchema.JSONSchema4[] {
   const selector: JSONSchemaProperties = {
     filter: {
-      type: 'string',
-      minLength: 1,
+      oneOf: [
+        {
+          type: 'string',
+          minLength: 1,
+        },
+        {
+          type: 'object',
+          properties: {
+            regex: {
+              type: 'string',
+            },
+            match: {
+              type: 'boolean',
+            },
+          },
+          required: ['regex', 'match'],
+        },
```

Comment:
nit - could you store this in a variable and share it between `custom` and filter`?